### PR TITLE
Introduce go blob-put

### PR DESCRIPTION
**Pull request by rafael-atticlabs**

### CODE REVIEW

**Comment by aboodman** [comment]
If it isn't too much extra trouble, can you please put this at `noms blob put` instead?

File: samples/go/blob-put/blob_put.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
```
Comment (cmasone-attic) [inline samples/go/blob-put/blob_put.go]: whitespace

File: samples/go/blob-put/blob_put.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	for i := 0; i < len(readers); i++ {
+		r, err := os.Open(filePath)
+		d.CheckErrorNoUsage(err)
+		defer r.Close()
+		r.Seek(int64(i)*chunkSize, 0)
+		limit := chunkSize
+		if i == len(readers)-1 {
+			limit += fileSize % chunkSize
+		}
+		lr := io.LimitReader(r, limit) // lastCh
```
Comment (cmasone-attic) [inline samples/go/blob-put/blob_put.go]: WHat's this comment about?

File: samples/go/blob-put/blob_put.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	cfg := config.NewResolver()
+	db, ds, err := cfg.GetDataset(flag.Arg(1))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not create dataset: %s\n", err)
+		return
+	}
+	defer db.Close()
+
+	profile.RegisterProfileFlags(flag.CommandLine)
```
Comment (cmasone-attic) [inline samples/go/blob-put/blob_put.go]: Already do this above

File: samples/go/blob-put/blob_put.go (hunk 2 above)
Comment (rafael-atticlabs) [inline samples/go/blob-put/blob_put.go]: woops. that should be on the limit line above.

File: samples/go/blob-put/blob_put.go (hunk 3 above)
Comment (rafael-atticlabs) [inline samples/go/blob-put/blob_put.go]: done

File: samples/go/blob-put/blob_put.go (hunk 1 above)
Comment (rafael-atticlabs) [inline samples/go/blob-put/blob_put.go]: done

**Pull request merged**